clinica/iotools/converters/aibl_to_bids/utils/clinical.py

```diff
@@ -334,64 +344,192 @@ def _get_csv_files_for_alternative_exam_date(
+            scans_dict[bids_id][session] = {
+                "T1/DWI/fMRI/FMAP": {},
+                Tracer.PIB: {},
+                Tracer.AV45: {},
+                Tracer.FMM: {},
+            }
+    return scans_dict
+
+
+def _format_time(time: str) -> str:
+    import datetime
+
+    date_obj = datetime.datetime.strptime(time, "%m/%d/%Y")
+    return date_obj.strftime("%Y-%m-%dT%H:%M:%S")
+
+
+def create_scans_dict(
+    clinical_data_dir: Path,
+    clinical_specifications_folder: Path,
+    bids_path: Path,
+) -> dict:
+    """Create a dictionary containing metadata per subject/session to write in scans.tsv.
+
+    Parameters
+    ----------
+    clinical_data_dir : Path
+        The path to the directory where the clinical data are stored.
+
+    clinical_specifications_folder : Path
+        The path to the folder containing the clinical specification files.
+
+    bids_path: Path
+        The path to the BIDS directory.
+
+    Returns
+    -------
+    pd.DataFrame :
+        A pandas DataFrame that contains the scans information for all sessions of all participants.
+    """
+    from clinica.iotools.converter_utils import viscode_to_session
+
+    scans_dict = _init_scans_dict(bids_path)
+
+    study = StudyName.AIBL.value
+
+    scans_specs = _load_specifications(clinical_specifications_folder, "scans.tsv")[
+        [study, f"{study} location", "BIDS CLINICA", "Modalities related"]
+    ].dropna()
+
+    for _, row in scans_specs.iterrows():
+        on_bad_lines = (
+            _flutemeta_badline if "flutemeta" in row[f"{study} location"] else "error"
+        )
+        file = _load_metadata_from_pattern(
+            clinical_data_dir, row[f"{study} location"], on_bad_lines
+        )
+        file["session_id"] = file["VISCODE"].apply(lambda x: viscode_to_session(x))
+
+        for bids_id in scans_dict.keys():
+            original_id = bids_id_factory(StudyName.AIBL)(
+                bids_id
+            ).to_original_study_id()
+            for session in scans_dict[bids_id].keys():
+                try:
+                    value = file[
+                        (file["RID"] == int(original_id))
+                        & (file["session_id"] == session)
+                    ][row[study]].item()
+                    if value == "-4":
+                        value = "n/a"
+                    elif row["BIDS CLINICA"] == "acq_time":
+                        value = _format_time(value)
+                except ValueError:
+                    value = "n/a"
+
+                modality = scans_dict[bids_id][session][row["Modalities related"]]
+
+                # Avoid writing over in case of modality "T1/..." because it is used twice
+                if (row["BIDS CLINICA"] not in modality) or (
+                    modality[row["BIDS CLINICA"]] == "n/a"
+                ):
+                    modality[row["BIDS CLINICA"]] = value
+    return scans_dict
+
+
+def write_scans_tsv(bids_dir: Path, scans_dict: dict) -> None:
```

Comment:
Either these are private, or `__all__` needs to be updated to reflect the new public objects of this module, right ? It basically depends if you are using these functions outside of this module.